cmd/kubeadm/app/apis/kubeadm/v1alpha3/conversion.go

```diff
@@ -103,6 +108,14 @@ func Convert_v1alpha3_ClusterConfiguration_To_kubeadm_ClusterConfiguration(in *C
 		return err
 	}
 
+	if len(in.UnifiedControlPlaneImage) == 0 {
+		out.UseHyperKubeImage = false
+	} else if strings.Contains(in.UnifiedControlPlaneImage, "/hyperkube:") {
```

Comment:
This check is weak and can lead to unsupported behaviours:

What if the user is using a custom image repository? This will get switched to a different image repository
Same questions about image tag

However, from the other side I don't think we should make this check too complex, so I will be fine with whatever check we agree upon

@neolit123 @timothysc opinions?

Reply:
the string "hypercube" has to be moved to constants.go.
the tag and repo conversation might be a bit too much, i think it's OK to have it like that.

we can also show a warning here too, right?